### PR TITLE
Replacing "q" calls with "DBA" calls

### PR DESCRIPTION
**Pull request by annando**
Several "q" calls got replaced. There are still some left, but that can be done in a following PR.

### CODE REVIEW

File: pumpio/pumpio.php [hunk 1]
```diff
@@ -681,16 +680,16 @@ function pumpio_sync(App $a)
...
-	if (DBA::isResult($r)) {
-		foreach ($r as $rr) {
-			Logger::log('pumpio: mirroring user '.$rr['uid']);
+	$pconfigs = DBA::selectToArray('pconfig', ['cat' => 'pumpio', 'k' => 'mirror', 'v' => '1']);
+	if (DBA::isResult($pconfigs)) {
+		foreach ($pconfigs as $rr) {
+			Logger::notice('pumpio: mirroring user '.$rr['uid']);
 			pumpio_fetchtimeline($a, $rr['uid']);
 		}
 	}
```
Comment (MrPetovan) [inline pumpio/pumpio.php]:
```suggestion
	foreach ($pconfigs as $rr) {
		Logger::notice('pumpio: mirroring user '.$rr['uid']);
		pumpio_fetchtimeline($a, $rr['uid']);
	}
```

File: pumpio/pumpio.php [hunk 2]
```diff
@@ -702,18 +701,17 @@ function pumpio_sync(App $a)
 
 	$abandon_limit = date(DateTimeFormat::MYSQL, time() - $abandon_days * 86400);
 
-	$r = q("SELECT * FROM `pconfig` WHERE `cat` = 'pumpio' AND `k` = 'import' AND `v` = '1' ORDER BY RAND() ");
-	if (DBA::isResult($r)) {
-		foreach ($r as $rr) {
+	$pconfigs = DBA::selectToArray('pconfig', ['cat' => 'pumpio', 'k' => 'import', 'v' => '1']);
+	if (DBA::isResult($pconfigs)) {
+		foreach ($pconfigs as $rr) {
```
Comment (MrPetovan) [inline pumpio/pumpio.php]:
```suggestion
	foreach ($pconfigs as $rr) {
```

File: pumpio/pumpio.php [hunk 3]
```diff
@@ -731,7 +729,7 @@ function pumpio_sync(App $a)
 		}
 	}
```
Comment (MrPetovan) [inline pumpio/pumpio.php:726]:
```suggestion
	}
```

File: windowsphonepush/windowsphonepush.php [hunk 4]
```diff
@@ -135,90 +135,88 @@ function windowsphonepush_settings(&$a, &$s)
...
-						break;
-					case "Dropped":
-						// mostly combines with Expired, in that case Device-URL will be deleted from pconfig (function send_push)
-						break;
-					default:
-						// error, mostly called by "" which means that the url (not "" which has been checked)
-						// didn't not received Microsoft Notification Server -> wrong url
-						Logger::log("ERROR: specified Device-URL '" . $device_url . "' didn't produced any response.");
-				}
+	$pconfigs = DBA::selectToArray('pconfig', ['cat' => 'windowsphonepush', 'k' => 'enable', 'v' => true]);
```
Comment (MrPetovan) [inline windowsphonepush/windowsphonepush.php]: Field list is missing.
```suggestion
	$pconfigs = DBA::selectToArray('pconfig', ['uid'], ['cat' => 'windowsphonepush', 'k' => 'enable', 'v' => true]);
```

File: windowsphonepush/windowsphonepush.php [hunk 5]
```diff
@@ -393,28 +391,23 @@ function windowsphonepush_updatesettings()
...
-	$r = q("SELECT * FROM `pconfig` WHERE `uid` <> " . local_user() . " AND
-						`cat` = 'windowsphonepush' AND
-						`k` = 'device_url' AND
-						`v` = '" . $device_url . "'");
-	if (count($r)) {
-		foreach ($r as $rr) {
-			DI::pConfig()->set($rr['uid'], 'windowsphonepush', 'device_url', '');
-			Logger::log("WARN: the sent URL was already registered with user '" . $rr['uid'] . "'. Deleted for this user as we expect to be correct now for user '" . local_user() . "'.");
-		}
+	$pconfigs = DBA::selectToArray('pconfig', ["`uid` != ? AND `cat` = ? AND `k` = ? AND `v` = ?", local_user(), 'windowsphonepush', 'device_url', $device_url]);
```
Comment (MrPetovan) [inline windowsphonepush/windowsphonepush.php]:
```suggestion
	$pconfigs = DBA::selectToArray('pconfig', ['uid'], ["`uid` != ? AND `cat` = ? AND `k` = ? AND `v` = ?", local_user(), 'windowsphonepush', 'device_url', $device_url]);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by annando** [comment]
Changes are applied.

File: pumpio/pumpio.php [hunk 6]
```diff
@@ -702,8 +700,7 @@ function pumpio_sync(App $a)
 	$abandon_limit = date(DateTimeFormat::MYSQL, time() - $abandon_days * 86400);
 
 	$pconfigs = DBA::selectToArray('pconfig', ['cat' => 'pumpio', 'k' => 'import', 'v' => '1']);
-	if (DBA::isResult($pconfigs)) {
-		foreach ($pconfigs as $rr) {
+	foreach ($pconfigs as $rr) {
 			if ($abandon_days != 0) {
 				if (DBA::exists('user', ["uid = ? AND `login_date` >= ?", $rr['uid'], $abandon_limit])) {
 					Logger::notice('abandoned account: timeline from user '.$rr['uid'].' will not be imported');
```
Comment (MrPetovan) [inline pumpio/pumpio.php]: Indentation.
```suggestion
		if ($abandon_days != 0) {
			if (DBA::exists('user', ["uid = ? AND `login_date` >= ?", $rr['uid'], $abandon_limit])) {
				Logger::notice('abandoned account: timeline from user '.$rr['uid'].' will not be imported');
```

File: pumpio/pumpio.php [hunk 7]
```diff
@@ -726,7 +723,6 @@ function pumpio_sync(App $a)
 				pumpio_getallusers($a, $rr["uid"]);
 				DI::pConfig()->set($rr['uid'], 'pumpio', 'contact_check', time());
 			}
```
Comment (MrPetovan) [inline pumpio/pumpio.php]:
```suggestion
			pumpio_getallusers($a, $rr["uid"]);
			DI::pConfig()->set($rr['uid'], 'pumpio', 'contact_check', time());
		}
```

File: mailstream/mailstream.php [hunk 8]
```diff
@@ -444,19 +439,7 @@ function mailstream_html_wrap(&$text)
...
-   `mailstream_item`
-WHERE
-  `mailstream_item`.`completed` IS NULL
-
-EOT;
-	$ms_item_ids = q($query);
+	$ms_item_ids = DBA::selectToArray('mailstream_item', ['message-id', 'uri', 'uid', 'contact-id'], ["`mailstream_item`.`completed` IS NULL"]);
 	if (DBA::isResult($ms_item_ids)) {
 		Logger::debug('mailstream_convert_table_entries processing ' . count($ms_item_ids) . ' items');
 		foreach ($ms_item_ids as $ms_item_id) {
```
Comment (MrPetovan) [inline mailstream/mailstream.php:445]:
```suggestion
	$ms_item_ids = DBA::selectToArray('mailstream_item', [], ['message-id', 'uri', 'uid', 'contact-id'], ["`mailstream_item`.`completed` IS NULL"]);
	Logger::debug('mailstream_convert_table_entries processing ' . count($ms_item_ids) . ' items');
	foreach ($ms_item_ids as $ms_item_id) {
```

File: mailstream/mailstream.php [hunk 9]
```diff
@@ -474,7 +457,7 @@ function mailstream_convert_table_entries()
 			Hook::fork(PRIORITY_LOW, 'mailstream_send_hook', $send_hook_data);
 		}
 	}
```
Comment (MrPetovan) [inline mailstream/mailstream.php]:
```suggestion
		Hook::fork(PRIORITY_LOW, 'mailstream_send_hook', $send_hook_data);
	}
```

File: pumpio/pumpio.php [hunk 10]
```diff
@@ -681,18 +680,16 @@ function pumpio_sync(App $a)
...
-	Logger::log('pumpio: cron_start');
+	Logger::notice('pumpio: cron_start');
 
-	$r = q("SELECT * FROM `pconfig` WHERE `cat` = 'pumpio' AND `k` = 'mirror' AND `v` = '1' ORDER BY RAND() ");
-	if (DBA::isResult($r)) {
-		foreach ($r as $rr) {
-			Logger::log('pumpio: mirroring user '.$rr['uid']);
-			pumpio_fetchtimeline($a, $rr['uid']);
-		}
+	$pconfigs = DBA::selectToArray('pconfig', ['cat' => 'pumpio', 'k' => 'mirror', 'v' => '1']);
```
Comment (MrPetovan) [inline pumpio/pumpio.php]:
```suggestion
	$pconfigs = DBA::selectToArray('pconfig', ['uid'], ['cat' => 'pumpio', 'k' => 'mirror', 'v' => '1']);
```

File: pumpio/pumpio.php [hunk 11]
```diff
@@ -702,18 +699,16 @@ function pumpio_sync(App $a)
 
 	$abandon_limit = date(DateTimeFormat::MYSQL, time() - $abandon_days * 86400);
 
-	$r = q("SELECT * FROM `pconfig` WHERE `cat` = 'pumpio' AND `k` = 'import' AND `v` = '1' ORDER BY RAND() ");
-	if (DBA::isResult($r)) {
-		foreach ($r as $rr) {
+	$pconfigs = DBA::selectToArray('pconfig', ['cat' => 'pumpio', 'k' => 'import', 'v' => '1']);
```
Comment (MrPetovan) [inline pumpio/pumpio.php]:
```suggestion
	$pconfigs = DBA::selectToArray('pconfig', ['uid'], ['cat' => 'pumpio', 'k' => 'import', 'v' => '1']);
```

File: mailstream/mailstream.php [hunk 12]
```diff
@@ -444,19 +439,7 @@ function mailstream_html_wrap(&$text)
...
-  `uid`,
-  `contact-id`
-FROM
-   `mailstream_item`
-WHERE
-  `mailstream_item`.`completed` IS NULL
-
-EOT;
-	$ms_item_ids = q($query);
+	$ms_item_ids = DBA::selectToArray('mailstream_item', ['message-id', 'uri', 'uid', 'contact-id'], ["`mailstream_item`.`completed` IS NULL"]);
```
Comment (MrPetovan) [inline mailstream/mailstream.php:442]:
```suggestion
	$ms_item_ids = DBA::selectToArray('mailstream_item', ['uid'], ['message-id', 'uri', 'uid', 'contact-id'], ["`mailstream_item`.`completed` IS NULL"]);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mailstream/mailstream.php (hunk 12 above)
Comment (annando) [reply mailstream/mailstream.php:442]: No, that's okay here. The uid mustn't added here.

File: mailstream/mailstream.php (hunk 8 above)
Comment (MrPetovan) [inline mailstream/mailstream.php]:
```suggestion
	Logger::debug('mailstream_convert_table_entries processing ' . count($ms_item_ids) . ' items');
	foreach ($ms_item_ids as $ms_item_id) {
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mailstream/mailstream.php [hunk 13]
```diff
@@ -300,12 +300,8 @@ function mailstream_subject($item)
...
-		"SELECT * FROM `contact` WHERE `id` = %d AND `uid` = %d",
-		intval($item['contact-id']),
-		intval($item['uid'])
-	);
-	if (!DBA::isResult($r)) {
+	$contact = Contact::selectFirst([], ['id' => $item['contact-id'], 'uid' => $item['uid']]);
+	if (!DBA::isResult($contact)) {
 			Logger::error(
 				'mailstream_subject no contact for item',
 				['id' => $item['id'],
```
Comment (MrPetovan) [inline mailstream/mailstream.php]: Indentation

File: mailstream/mailstream.php [hunk 14]
```diff
@@ -315,7 +311,6 @@ function mailstream_subject($item)
 			);
```
Comment (MrPetovan) [inline mailstream/mailstream.php]: Indentation

File: mailstream/mailstream.php [hunk 15]
```diff
@@ -444,22 +439,9 @@ function mailstream_html_wrap(&$text)
...
-	$ms_item_ids = q($query);
-	if (DBA::isResult($ms_item_ids)) {
-		Logger::debug('mailstream_convert_table_entries processing ' . count($ms_item_ids) . ' items');
-		foreach ($ms_item_ids as $ms_item_id) {
+	$ms_item_ids = DBA::selectToArray('mailstream_item', [], ['message-id', 'uri', 'uid', 'contact-id'], ["`mailstream_item`.`completed` IS NULL"]);
+	Logger::debug('mailstream_convert_table_entries processing ' . count($ms_item_ids) . ' items');
+	foreach ($ms_item_ids as $ms_item_id) {
 			$send_hook_data = array('uid' => $ms_item_id['uid'],
 						'contact-id' => $ms_item_id['contact-id'],
 						'uri' => $ms_item_id['uri'],
```
Comment (MrPetovan) [inline mailstream/mailstream.php]: Indentation

File: mailstream/mailstream.php [hunk 16]
```diff
@@ -471,10 +453,9 @@ function mailstream_convert_table_entries()
 								continue;
 			}
 			Logger::info('mailstream_convert_table_entries: convert item to workerqueue', $send_hook_data);
```
Comment (MrPetovan) [inline mailstream/mailstream.php]: Indentation

File: pumpio/pumpio.php [hunk 17]
```diff
@@ -702,18 +699,16 @@ function pumpio_sync(App $a)
...
+					Logger::notice('abandoned account: timeline from user '.$rr['uid'].' will not be imported');
 					continue;
 				}
 			}
 
-			Logger::log('pumpio: importing timeline from user '.$rr['uid']);
+			Logger::notice('pumpio: importing timeline from user '.$rr['uid']);
 			pumpio_fetchinbox($a, $rr['uid']);
 
 			// check for new contacts once a day
```
Comment (MrPetovan) [inline pumpio/pumpio.php]: Indentation

File: pumpio/pumpio.php [hunk 18]
```diff
@@ -728,10 +723,9 @@ function pumpio_sync(App $a)
 				pumpio_getallusers($a, $rr["uid"]);
 				DI::pConfig()->set($rr['uid'], 'pumpio', 'contact_check', time());
 			}
```
Comment (MrPetovan) [inline pumpio/pumpio.php]: Indentation

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: pumpio/pumpio.php (hunk 6 above)
Comment (annando) [reply pumpio/pumpio.php]: Fixed

**Comment by annando** [comment]
@MrPetovan these addons are a bunch of ugly formatted lines of code. I'm unsure if we really should fix all these issues with this PR. Because otherwise I had to reformat every file I guess.

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
I feel like it's a matter of "you touch it, you fix it", but it is a big involvement sometimes.

**Pull request merged**